sympy/core/mul.py

```
@@ -1541,9 +1541,15 @@ def _eval_is_imaginary(self):
             return self._eval_real_imag(False)
 
     def _eval_is_hermitian(self):
+        from sympy.simplify import simplify
+        if simplify(self.func(*self.args) - self._eval_adjoint()).is_zero:
```

`simplify` has potentially very long evaluation time, I am not sure this is how other similar methods behave, rather than failing early.

---

I used simplify because simplify is used here:
https://github.com/sympy/sympy/blob/a6f35dbfa24ba66ff8f664a7fae7011a435f0832/sympy/matrices/matrixbase.py#L1331-L1333
I think that `simp_func` refers to `simplify`, but maybe these two methods are not that similar.

---

Digging deeper, I became quite sure that automatic simplification is undesirable here. Consider that `sympy.Expr._eval_transpose` contains a check for `is_hermitian`. That would mean that transposing a matrix with a complex expression containing complicated expressions with noncommutative entries will effectively stop working.

---

The idea with that code is that a user can pass in their own function for simplifying things. It is not a good model to base other things on.

Generally simplify should be avoided within the sympy codebase because it is both slow and heuristic making things not well defined. Usually the suggestion instead is either to use a more targeted simplification function or just not to use any simplification.

---

I checked the code for matrix property testing. All public methods *ask* the user whether to simplify in addition. These methods also aren't called by sympy internally, as indicated by the comment right before the code you quoted.

---

`self.func(*self.args) - self._eval_adjoint()` automatically simplifies to zero — at least in the cases currently being tested. I'm not sure what cases if any were being handled because of the call to simplify. If you had some cases that could only be handled with the call to simplify in mind, they should probably be added as tests.

---

> self.func(*self.args) - self._eval_adjoint() automatically simplifies to zero

I agree with you, I think that simplify was not needed here.